### PR TITLE
Adds support for building locally

### PR DESCRIPTION
**Pull request by Bucknalla**
Change-type: minor
Signed-off-by: Alex Bucknall <alex.bucknall@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,27 +2,30 @@
 
 > The `create-react-app` version of building docusaurus websites
 
-This is the official Docusaurus theme for balena docs. [Docusaurus 2](https://docusaurus.io/)is a modern static website generator that we used to generate docs. This repository hosts a customized docusarus theme and a docusaurus builder GitHub action. 
+This is the official Docusaurus theme for balena docs.
```
Comment (vipulgupta2048) [inline README.md:5]: Are you running an auto linter on markdown? It feels like it's adding a lot of changes that aren't relevant to the PR. Do reset them and add only the local preview section

File: README.md [hunk 2]
```diff
@@ -43,41 +46,59 @@ jobs:
...
-To preview how your website will end up looking, make your PR on a docusaurus-enabled repository and the preview URL will be generated. 
-To add new features to the theme, you can start contributing here!
+To develop your site locally you should to have the following installed:
 
+- [`Docker`](https://docs.docker.com/get-docker/)
+
+With `Docker` installed, open a shell in your terminal, navigate to the root of your project's repo and run the following command:
+
+```bash
+docker run -t -i -e "DEV=true" -v $(pwd):/app/local -p 80:80 product:latest
```
Comment (vipulgupta2048) [inline README.md:97]:
```suggestion
docker run -t -i -e "DEV=true" -v $(pwd):/app/local product-os/docusaurus-builder:latest
```

I am assuming you will be changing the name of the image once it's published

File: README.md [hunk 3]
```diff
@@ -43,41 +46,59 @@ jobs:
...
+```
+
+> Note - if you are trying to run this from Windows, please use powershell
```
Comment (vipulgupta2048) [inline README.md:100]: I would add a "Why" to this statement to let the user know why you need recommend that

File: README.md [hunk 4]
```diff
@@ -43,41 +46,59 @@ jobs:
...
+
+Once you see `[SUCCESS] Docusaurus website is running at: http://localhost:80/` in your terminal, your site will be available to your web browser at `http://localhost:80`.
```
Comment (vipulgupta2048) [inline README.md:102]:
```suggestion
Once you see a success message in your terminal, your docusaurus site will be available on [http://localhost:80](http://localhost:80).
```

I would also recommend adding that it won't hot reload or watch for changes. Instead users will need to run this command again to regenerate their preview site each time to test their changes and get a new preview site

File: README.md (hunk 1 above)
Comment (Bucknalla) [reply README.md:5]: I like my auto linting :P

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Test
```
Comment (vipulgupta2048) [inline docs/README.md:1]: Remove this file?

File: entrypoint.sh [hunk 6]
```diff
@@ -1,8 +1,23 @@
 #!/bin/sh -l
 
-echo "Inputs: $1 $2 $3 $4"
-. /app/docusaurusify.sh $1 $2 $3 $4
+if [ "$DEV" = 'true' ] ; then
+    echo "Development mode is enabled"
+    export PROJECT_NAME="dev build"
```
Comment (vipulgupta2048) [inline entrypoint.sh:5]:
```suggestion
    export PROJECT_NAME="Development build"
```

**Review by vipulgupta2048 — COMMENTED** [review]
LGTM, once Flowzone is tested to not break.

File: README.md (hunk 1 above)
Comment (vipulgupta2048) [reply README.md:5]: But, the git history won't like the noise when that shows up on blame. To have a clean git history I recommend, linting these preferably in a different PR

File: docs/README.md (hunk 5 above)
Comment (Bucknalla) [reply docs/README.md]: It's useful for locally testing the builder, I'll keep it but adjust it

**Comment by klutchell** [comment]
@Bucknalla I suggest adding a docker-bake file to this PR to enable arm64 builds.

```hcl
# docker-bake.hcl
target "default" {
  platforms = [
    "linux/amd64",
    "linux/arm64"
  ]
}
```

**Comment by Bucknalla** [comment]
LGTM

**Pull request merged**